server/src/utils/pascal-case.ts

```diff
@@ -0,0 +1,32 @@
+import isObject from 'lodash.isobject';
+import lodashCamelCase from 'lodash.camelcase';
+import { PascalCase, PascalCasedPropertiesDeep } from 'type-fest';
+
+export const capitalizeFirstLetter = (str: string) => {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+};
+
+export const pascalCase = <T>(text: T) =>
+  capitalizeFirstLetter(
+    lodashCamelCase(text as unknown as string),
+  ) as PascalCase<T>;
+
+export const pascalCaseDeep = <T>(value: T): PascalCasedPropertiesDeep<T> => {
```

Comment:
I don't understand all the "as" typing! why is it needed?

Reply:
It's to ensure typing is also PascalCase, by example for the keys of a constant object.
Lodash otherwise return a classic object type